apps/subscription.go

```diff
@@ -22,48 +22,48 @@ const (
 	//   Requires: model.PermissionViewMembers.
 	SubjectUserCreated Subject = "user_created"
 
-	// SubjectUserJoinedChannel, SubjectUserLeftChannel watch the specified
-	// channel for users joining and leaving it.
+	// SubjectUserJoinedChannel, SubjectUserLeftChannel behave differently
+	// depending on the provided parameters. If no channel is specified, the
+	// subscription is system-wide, and the event is fired for every channel the
+	// subscriber joins or leaves. If a channel is specified (and the user has
+	// access to the channel), the event is fired for all users joining or
+	// leaving the channel.
```

Comment:
I'm not 100% a fan of the duality of the subject, but the previous `Self` subject was even more confusing. I guess it's fine.

If anyone has better ideas on how to structure the subject, please comment!

Reply:
I think it would be nice to have a difference between `SubjectUserCreated` and `SystemSubjectUserCreated` or something like that. It could be a good idea to create a Jira ticket to explore this more in depth IMO.